hyperjumptech/monika
# feat(native fetch): Implement Node.js fetch

# Monika Pull Request (PR)
This PR resolves #1241 

## What feature/issue does this PR add

1. Implement Node.js fetch under `--native-fetch`

## How did you implement / how did you fix it

1. Change flag `--experimental-fetch` to `--native-fetch`
2. Implement Node.js fetch under flag `--native-fetch`
3. Change typings from `AxiosRequestHeaders` to be compatible with plain TypeScript / fetch
4. Change typings from `AxiosRequestConfig` to be compatible with plain TypeScript / fetch
5. Generalize typing for sending HTTP request into one `type HttpRequestParams`
6. Map Node.js fetch result 

## How to test

1. Given below config
```
probes:
  - id: '1'
    name: GitHub
    description: Multiple
    interval: 5
    requests:
      - url: https://github.com
        timeout: 7000
      - url: https://github.com/hyperjump
        timeout: 7000
  - id: '2'
    name: HTML form submission
    description: simulate html form submission
    interval: 5
    requests:
      - url: https://httpbin.org/post
        method: POST
        headers:
          Accept: application/json
        body:
          username: someusername
          password: somepassword
```
2. Run `npm run start -- --native-fetch --verbose`
3. Observe log statements for `Probing URL with Node.js fetch`
![image](https://github.com/hyperjumptech/monika/assets/9563873/17c5a426-10c5-43d4-92ee-328a116bbcb8)
